aptos-move/e2e-testsuite/src/tests/automation_transaction.rs

```diff
@@ -0,0 +1,343 @@
+        )
+    }
+
+    fn create_automation_txn_with_arguments(
+        &self,
+        seq_num: u64,
+        inner_payload: EntryFunction,
+        expiry_time: u64,
+        max_gas_amount: u64,
+        gas_price_cap: u64,
+    ) -> SignedTransaction {
+        let txn_arguments = AutomationTransactionArguments::new(
+            inner_payload,
+            expiry_time,
+            max_gas_amount,
+            gas_price_cap,
+        );
+        let automation_txn = TransactionPayload::Automation(
+            AutomationTransactionPayload::EntryFunctionArguments(txn_arguments),
+        );
+        self.txn_sender
+            .account()
+            .transaction()
+            .payload(automation_txn)
+            .sequence_number(seq_num)
+            .sign()
+    }
+
+    fn create_automation_txn_with_raw_inner_payload(
+        &self,
+        seq_num: u64,
+        inner_entry_function_bytes: Vec<u8>,
+        expiry_time: u64,
+        max_gas_amount: u64,
+        gas_price_cap: u64,
+    ) -> SignedTransaction {
+        let automation_txn_payload = aptos_framework_sdk_builder::automation_registry_register(
+            inner_entry_function_bytes,
+            expiry_time,
+            max_gas_amount,
+            gas_price_cap,
+        )
+        .into_entry_function();
+        let automation_txn = TransactionPayload::Automation(
+            AutomationTransactionPayload::EntryFunction(automation_txn_payload),
+        );
+        self.txn_sender
+            .account()
+            .transaction()
+            .payload(automation_txn)
+            .sequence_number(seq_num)
+            .sign()
+    }
+
+    fn check_miscellaneous_output(output: TransactionOutput, expected_status_code: StatusCode) {
+        match output.status() {
+            TransactionStatus::Keep(ExecutionStatus::MiscellaneousError(maybe_status_code)) => {
+                assert_eq!(
+                    maybe_status_code.as_ref().unwrap(),
+                    &expected_status_code,
+                    "{output:?}"
+                );
+            },
+            _ => panic!("Unexpected transaction status: {output:?}"),
+        }
+    }
+}
+
+impl Deref for AutomationTransactionTestContext {
+    type Target = FakeExecutor;
+
+    fn deref(&self) -> &Self::Target {
+        &self.executor
+    }
+}
+
+impl DerefMut for AutomationTransactionTestContext {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.executor
+    }
+}
+
+#[test]
+fn check_successful_registration() {
+    let mut test_context = AutomationTransactionTestContext::new();
+    // Prepare inner-entry-function to be automated.
+    let dest_account = test_context.new_account_data(0, 0);
+    let inner_entry_function =
+        aptos_framework_sdk_builder::supra_coin_mint(dest_account.address().clone(), 100)
+            .into_entry_function();
+
+    let view_output = test_context.execute_view_function(
+        str::parse("0x1::timestamp::now_seconds").unwrap(),
```

Comment:
It is best to define constant literals and use it here instead of directly using the strings.